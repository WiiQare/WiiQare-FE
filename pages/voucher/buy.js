import { createContext, useState } from 'react';
import Head from 'next/head';
import { useRouter } from 'next/router';
import DashboardLayout from '../../layouts/Dashboard';
import CardHeader from '../../components/atoms/Card/Header';
import Stepper from '../../components/atoms/Stepper';
import Step from '../../components/atoms/Stepper/step';
export const FormContext = createContext();

const Page = () => {
  const { step, redirect_status, payment_intent } = useRouter().query;
  const [activeStepIndex, setActiveStepIndex] = useState(
    step == 'end' && redirect_status == 'succeeded' ? 2 : 0,
  );
  const [formData, setFormData] = useState({});

  return (
    <>
      <Head>
        <title>Envoyer un pass santé</title>
      </Head>
      <div className="p-2 space-y-6 md:py-8 md:px-6">
        <CardHeader
          title={'Envoyer un pass santé'}
          breadcrumbs={[
            {
              item: 'Accueil',
              link: '/',
            },
            {
              item: 'Envoyer un pass santé',
              link: '/voucher/buy',
            },
          ]}
          download={false}
          print={false}
        />

        <section className="w-full pb-20 md:pb-0">
          <div className="w-full overflow-hidden md:col-span-2 rounded-lg py-8 flex flex-col gap-6 bg-white drop-shadow-sm">
            <div className="px-8 space-y-8">
              <div>
                <h2 className="font-semibold text-3xl">
                  Envoyer un pass santé
                </h2>
              </div>
              <div className="flex flex-col gap-4">
                <FormContext.Provider
                  value={{
                    activeStepIndex,
                    setActiveStepIndex,
                    formData,
                    setFormData,
                    payment_intent,
                  }}
                >
                  <div className="flex flex-col items-center justify-start">
                    <Stepper />
                    <Step />
                  </div>
                </FormContext.Provider>
              </div>
            </div>
          </div>
        </section>
      </div>
    </>
  );
};

Page.getLayout = (page) => (
  <DashboardLayout className="space-y-8">{page}</DashboardLayout>
);

export default Page;
