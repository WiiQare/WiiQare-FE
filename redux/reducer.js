import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  client: { register: {}, patient: {} },
};

export const ReducerSlice = createSlice({
  name: 'wiiqare',
  initialState,
  reducers: {
    setRegister: (state, action) => {
      state.client.register = { ...action.payload };
    },
    setPatientDispatch: (state, action) => {
      state.client.patient = { ...action.payload };
    },
  },
});

export const { setRegister, setPatientDispatch } = ReducerSlice.actions;

export default ReducerSlice.reducer;
