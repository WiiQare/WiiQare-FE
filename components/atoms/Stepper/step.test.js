import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Step from './step';
import { FormContext } from '../../../pages/voucher/buy';
import { SessionProvider } from 'next-auth/react';
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';
import { QueryClient, QueryClientProvider } from 'react-query';

describe('Stepper/Step', () => {
  beforeEach(() => {
    console.log = jest.fn();
  });

  it('renders identity', () => {
    const { container } = render(
      <FormContext.Provider
        value={{
          activeStepIndex: 0, // identity step
        }}
      >
        <SessionProvider
          session={{
            user: {
              data: {
                userId: 1,
              },
            },
          }}
        >
          <Provider store={store}>
            <QueryClientProvider client={new QueryClient()}>
              <Step />
            </QueryClientProvider>
          </Provider>
        </SessionProvider>
      </FormContext.Provider>,
    );
    expect(container).toMatchSnapshot('identity step');
  });

  it.todo('renders payment');

  it('renders send', () => {
    const { container } = render(
      <FormContext.Provider
        value={{
          activeStepIndex: 2, // send step
        }}
      >
        <SessionProvider
          session={{
            user: {
              data: {
                userId: 1,
              },
            },
          }}
        >
          <Provider store={store}>
            <QueryClientProvider client={new QueryClient()}>
              <Step />
            </QueryClientProvider>
          </Provider>
        </SessionProvider>
      </FormContext.Provider>,
    );
    expect(container).toMatchSnapshot('send step');
  });
});
