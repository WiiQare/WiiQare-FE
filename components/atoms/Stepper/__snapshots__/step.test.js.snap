// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper/Step renders identity: identity step 1`] = `
<div>
  <div
    class="flex flex-col justify-start md:w-2/3 w-full gap-4"
  >
    <div
      class="mt-6"
    >
      <div
        class="w-full bg-gray-100 py-4 px-6 border-2 border-dashed border-gray-700 rounded-lg cursor-pointer hover:shadow-md hover:bg-gray-200 flex justify-between"
      >
        <div
          class="flex gap-5 items-center"
        >
          <svg
            fill="none"
            height="26"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H5a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="8.5"
              cy="7"
              r="4"
            />
            <line
              x1="20"
              x2="20"
              y1="8"
              y2="14"
            />
            <line
              x1="23"
              x2="17"
              y1="11"
              y2="11"
            />
          </svg>
          <h4
            class="font-semibold md:text-lg text-sm"
          >
            Ajouter un bénéficiaire
          </h4>
        </div>
        <svg
          fill="none"
          height="26"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="26"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Stepper/Step renders send: send step 1`] = `
<div>
  <div
    class="flex flex-col gap-6 items-center h-screen w-full overflow-hidden"
  >
    <svg
      aria-hidden="true"
      class="text-red-600"
      fill="currentColor"
      height="100"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 20 20"
      width="100"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
    <span>
      Error Voucher !
    </span>
  </div>
</div>
`;
